source/_posts/2019-02-04-introducing-home-assistant-cli.markdown

```diff
@@ -0,0 +1,180 @@
+---
+layout: post
+title: "Introducing Home Assistant CLI aka. hass-cli"
+description: "Interact with Home Assistant on a new level"
+date: 2019-02-04 00:01:00
+date_formatted: "February 4, 2018"
+author: Max Rydahl Andersen
+author_twitter: maxandersen
+comments: true
+categories: Announcements
+---
+
+When I started using Home Assistant (HA) I found the browser UI great for casual
+usecases. For my usecases it felt a bit slow when trying to debug/trace what
+was going on. I realized I was missing a way to interact with HA via a command
+line (CLI) with the power of auto-completion and direct access without having to wait
+for pages to load and clicking around.
+
+Thus I started working on [Home Assistant CLI][github-hass-cli], a project that
+was initially started by [Fabian Affolter][@fabaff].
+
+Last week we released version 0.4, which provides majority of the features I was
+looking for to do from a CLI.
+
+## Feature highlights:
+   
+   - Get configuration info (`hass-cli config`)
+   - List, get, edit and delete state for entities (`hass-cli entity`')
+   - Query history with relative time ('hass-cli entity history')
+   - List and run services (`hass-cli services`)
+   - Execute templates locally and remotely (`hass-cli template`)
+   - Control over columns, sorting, etc. (`hass-cli --columns attr1,attr2 --sort-by attr3`)
+   - Shell completion for most commands and arguments (`hass-cli completion zsh`)
+   - Get logs (`hass-cli system log`)
+   - Run discovery (`hass-cli discover`)
+   - Show map for zone (`hass-cli map`)
+   - Call raw api directly (`hass-cli raw`)
+   - ... and more
+
+## Usage
+
+For the basic intro to `hass-cli` see the docs at https://github.com/home-assistant/home-assistant-cli.
+
+## Installation
+
+Install latest `homeassistant-cli` by pip:
+
+    $ pip3 install homeassistant-cli
```

Comment:
We can also provide the upgrade instructions here.

Reply:
hmm - isn't pip3 using install for both install and upgrade ? 

in any case I've added a line with `pip3 install --upgrade` variation.